crate/server/src/config/params/server_params.rs

```diff
@@ -80,10 +81,32 @@ impl ServerParams {
             None
         };
 
+        let jwt_config = match (
+            &conf.auth.jwt_issuer_uri,
+            &conf.auth.jwks_uri,
+            &conf.auth.jwt_audience,
+        ) {
+            (Some(jwt_issuer_uri), Some(jwks_uri), jwt_audience) => {
+                let min_length = jwt_issuer_uri.len().min(jwks_uri.len());
+                let mut idp_configs = Vec::with_capacity(min_length);
+                for index in 0..min_length {
```

Comment:
this can be replaced by a `zip`:
```suggestion
                let idp_configs = jwt_issuer_uri.into_iter().zip(jwks_uri.into_iter()).enumerate().map(|(i, (issuer, jwks))| {...}).collect();
```

If you don't like the map you can still use the zip with a for loop.